File: Greedy/Greedy_2.java
import java.util.Scanner;
import java.util.Arrays;

public class Greedy_2 {
    public static void main(String[] args) throws Exception {

        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int M = sc.nextInt();

        int result = 10001;

        int arr[] = new int[M];
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < M; j++) {
                arr[j] = sc.nextInt();
            }
            Arrays.sort(arr);
            // 각 열의 가장 작은 수
            if (i == 0 || result < arr[0]) {
                result = arr[0];
            }
        }

        System.out.println(result);
    }
}

File: Greedy/Greedy_2.py
# N: 행, M: 열
N, M = map(int,input().split())

result = 0
# for (반복문)
# for 변수 in 리스트(또는 튜플, 문자열...)
# range(1,11) 으로 숫자리스트를 만들어 줄 수도 있다 (1이상 11미만)
for i in range(N):
    listM = list(map(int,input().split(" ")))
    listM.sort()
    # if elif (조건문)
    if i == 1:
        result = listM[0]
    elif result < listM[0]:
        result = listM[0]

print(result)
